test/Tests/src/Data/Vector_Spec.enso

```diff
@@ -257,7 +257,9 @@ spec = Test.group "Vectors" <|
         [1, 3, 1, 2, 2, 1].distinct . should_equal [1, 3, 2]
         ["a", "a", "a"].distinct . should_equal ["a"]
         [].distinct . should_equal []
+        [1, 1.0, 2, 2.0].distinct . should_equal [1, 2]
```

Comment:
Shouldn't this last case actually go to the mixed type group below?

It's unclear - on one hand both `1` and `1.0` have the common supertype `Number` but still, they are mixed types after all.